db/migrate/20150213162341_add_value_constraints_batch_1.rb

```diff
@@ -0,0 +1,1194 @@
+
+ALTER TABLE inputs ALTER COLUMN name SET NOT NULL,
+                   ADD CONSTRAINT normalized_input_name CHECK (is_whitespace_normalized(name));
+ALTER TABLE inputs ALTER COLUMN access_level SET DATA TYPE level_of_access,
+                   ALTER COLUMN access_level SET DEFAULT 4;
+
+-- add after cleanup:
+/* ALTER TABLE inputs ALTER COLUMN format_id SET NOT NULL; */
+
+/* LIKELIHOODS */
+
+-- no current constraints other than those in AddUniquenessConstraints migration
+
+
+/* MACHINES */
+
+ALTER TABLE machines ADD CONSTRAINT well_formed_machine_hostname CHECK (is_host_address(hostname));
+-- decide on additional constraints (if any)
+
+/* MANAGEMENTS */
+
+-- possibly:
+-- ALTER TABLE managements ALTER COLUMN citation_id SET NOT NULL;
+
+-- show null dateloc counts:
+--   SELECT COUNT(*), date FROM managements WHERE dateloc IS NULL GROUP BY date;
+/* ALTER TABLE managements ALTER COLUMN dateloc SET NOT NULL; */
+-- decide on other date, dateloc constraints
+
+ALTER TABLE managements ALTER COLUMN mgmttype SET NOT NULL;
+-- get a count of unrecognized mgmttype values:
+-- SELECT COUNT(*) FROM managements WHERE mgmttype NOT IN ( 'burned', 'coppice', 'cultivated', 'cultivated or grazed', 'fertilization_Ca', 'fertilization_K', 'fertilization_N', 'fertilization_P', 'fertilization_other', 'fungicide', 'grazed', 'harvest', 'herbicide', 'irrigation', 'light', 'pesticide', 'planting (plants / m2)', 'row spacing', 'seeding', 'tillage','warming_soil','warming_air','initiation of natural succession','major storm','root exclusion', 'trenching', 'CO2 fumigation', 'soil disturbance', 'rain exclusion');
+-- show what these values are:
+-- SELECT DISTINCT mgmttype FROM managements WHERE mgmttype NOT IN ( 'burned', 'coppice', 'cultivated', 'cultivated or grazed', 'fertilization_Ca', 'fertilization_K', 'fertilization_N', 'fertilization_P', 'fertilization_other', 'fungicide', 'grazed', 'harvest', 'herbicide', 'irrigation', 'light', 'pesticide', 'planting (plants / m2)', 'row spacing', 'seeding', 'tillage','warming_soil','warming_air','initiation of natural succession','major storm','root exclusion', 'trenching', 'CO2 fumigation', 'soil disturbance', 'rain exclusion');
+
+-- decide on level and units constraints
+
+ALTER TABLE managements ALTER COLUMN notes SET NOT NULL,
+                        ALTER COLUMN notes SET DEFAULT '';
+
+
+/* METHODS */
+
+ALTER TABLE methods ALTER COLUMN name SET NOT NULL;
+ALTER TABLE methods ADD CONSTRAINT normalized_method_name CHECK (is_whitespace_normalized(name));
+ALTER TABLE methods ALTER COLUMN description SET NOT NULL;
+-- Additional constraints will be added when candidate key is implemented.
+
+
+/* MIMETYPES */
+
+-- Constraints are in AddUniquenessConstraints migration.
+
+
+/* MODELS */
+
+ALTER TABLE models ALTER COLUMN model_name SET NOT NULL,
+                   ADD CONSTRAINT no_spaces_in_model_name CHECK (is_whitespace_free(model_name));
+ALTER TABLE models ALTER COLUMN revision SET NOT NULL,
```

Comment:
Currently these are all numbers, the empty string or 'unk'.  But we can relax this restriction if you like ( whitespace-normalized?).  Say the word.